src/tree/updater_gpu_hist.cu

```diff
@@ -1444,23 +1455,101 @@ class GPUHistMakerSpecialised{
                                           column_sampling_seed));
         });
 
-    // Find the cuts.
+    // Initialize Sketches across all batches
+    common::SketchContainer sketch_container;
+    sketch_container.sketches_.resize(info_->num_col_);
+    sketch_container.col_locks_.resize(info_->num_col_);
+#pragma omp parallel for schedule(static)
```

Comment:
This is a fairly simple initialization. Does it need OpenMP parallelization?

Reply:
i was trying to speed this up for wider datasets. i have changed this now to parallelize only when the number of features are > 1000. the openmp threads should be pre-created - right? i presume they aren't created/destroyed during every one of these parallel blocks.